timemachine/cpp/src/nonbonded_interaction_group.hpp

```diff
@@ -4,16 +4,18 @@
 #include "neighborlist.hpp"
 #include "potential.hpp"
 #include <array>
+#include <optional>
 #include <vector>
 
 namespace timemachine {
 
 template <typename RealType> class NonbondedInteractionGroup : public Potential {
 
 private:
-    const int N_; // N_ = NC_ + NR_
+    const int N_; // total number of atoms, i.e. first dimension of input coords, params
     int NR_;      // number of row atoms
     int NC_;      // number of column atoms
+    int K_;       // number of interacting atoms, K_ = NC_ + NR_
```

Comment:
Can we turn this into a method? i.e. `get_num_interacting_atoms()` - generally we'd like to avoid creating another member that can be computed from two other members.

Reply:
(especially since `NR_` and `NC_` can change I think? and we'd like to avoid bookkeeping of keeping it synced s.t. `K_=NR_+NC_` for all values)